osu.Game/Overlays/Comments/CommentEditor.cs

```diff
@@ -37,6 +42,13 @@ public abstract partial class CommentEditor : CompositeDrawable
 
         protected TextBox TextBox { get; private set; } = null!;
 
+        protected readonly IBindable<APIUser> User = new Bindable<APIUser>();
+
+        [Resolved]
+        private IAPIProvider api { get; set; } = null!;
```

Comment:
several `CommentEditor` subclasses also resolve their own `IAPIProvider`. Should I resolve one `protected IAPIProvider API { get; private set; } = null!;` here instead and have the subclasses use that? Or is each subclass resolving its own ok?

Reply:
Better to resolve once and expose as protected. If you resolve as private in each subclass you end up having duplicated fields, which makes the objects bigger than they need to be (not by much, a reference is just an memory address, loosely speaking, but still).